spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -48,7 +48,21 @@
 
 abstract class BaseSparkAction<R> implements Action<R> {
 
-  protected abstract Table table();
+  private final SparkSession spark;
```

Comment:
Previously, we always assumed there is going to be a table we operate on. That's not the case with actions like snapshot and migrate. Plus, it does not make sense to init the Java context in all actions. We can do this in parent.